viewing_party/party.py

```diff
@@ -1,23 +1,165 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+
+    if title and genre and rating:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    else:
+        return None
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    watched_movie = {}
+
+    for list, movie_data in user_data.items():
+        for data in movie_data:
+            if data["title"] == title:
+                watched_movie = data
+                user_data["watchlist"].remove(data)
+
+    if len(watched_movie) >= 1:
+        add_to_watched(user_data, watched_movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    rating_sum = 0
+    rating_count = 0
+
+    for list, movie_data in user_data.items():

```

Comment:
Good job looking up different methods to use. I would also push you to think of your own implementation and get familiar with explaining how this method works under the hood.